### PR TITLE
Refactors logging in the csi packages

### PR DESCRIPTION
**Pull request by 0sewa0**
Moves logger to the `config.go` in each package, and have the logger on a per package level.
Also log messages should always start with lowercase.

### CODE REVIEW

File: controllers/csi/driver/server.go [hunk 1]
```diff
@@ -335,11 +306,11 @@ func (svr *CSIDriverServer) loadVolumeInfo(volumeID string) (*metadata.Volume, e
 	if volume == nil {
 		return &metadata.Volume{}, nil
 	}
-	svr.log.Info("loaded volume info", "id", volume.VolumeID, "pod name", volume.PodName, "version", volume.Version, "dynakube", volume.TenantUUID)
+	log.Info("loaded volume info", "id", volume.VolumeID, "pod name", volume.PodName, "version", volume.Version, "dynakube", volume.TenantUUID)
 	return volume, nil
 }
 
-func logGRPC(log logr.Logger) grpc.UnaryServerInterceptor {
```
Comment (chrismuellner) [inline controllers/csi/driver/server.go:342]: could we rework when this is logged as well?
I don't think we have to log every successful grpc call

File: controllers/csi/provisioner/reconciler.go [hunk 2]
```diff
@@ -86,8 +78,9 @@ func (r *OneAgentProvisioner) SetupWithManager(mgr ctrl.Manager) error {
 var _ reconcile.Reconciler = &OneAgentProvisioner{}
 
 func (r *OneAgentProvisioner) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	rlog := log.WithValues("namespace", request.Namespace, "name", request.Name)
-	rlog.Info("Reconciling DynaKube")
+	log = log.WithValues("namespace", request.Namespace, "dynakube", request.Name)
+	defer resetLogger()
```
Comment (chrismuellner) [inline controllers/csi/provisioner/reconciler.go:82]: we could explicitly log these values instead of having them on the package wide logger?

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: controllers/csi/driver/server.go [hunk 3]
```diff
@@ -335,23 +306,30 @@ func (svr *CSIDriverServer) loadVolumeInfo(volumeID string) (*metadata.Volume, e
...
 			return handler(ctx, req)
 		}
-
-		log.Info("GRPC call", "method", info.FullMethod, "request", req)
-
+		methodName := ""
+		if info.FullMethod == "/csi.v1.Node/NodePublishVolume" {
+			req := req.(csi.NodePublishVolumeRequest)
+			methodName = "NodePublishVolume"
+			log.Info("GRPC call", "method", info.FullMethod, "volume-id", req.VolumeId)
```
Comment (chrismuellner) [inline controllers/csi/driver/server.go:322]: wouldn't it be enough to log the `volume-id` and if it's node publish/unpublish here

File: controllers/csi/driver/server.go [hunk 4]
```diff
@@ -335,23 +306,30 @@ func (svr *CSIDriverServer) loadVolumeInfo(volumeID string) (*metadata.Volume, e
...
+			methodName = "NodeUnpublishVolume"
+			log.Info("GRPC call", "method", info.FullMethod, "volume-id", req.VolumeId)
+		}
 		resp, err := handler(ctx, req)
 		if err != nil {
-			log.Error(err, "GRPC call failed")
+			log.Error(err, fmt.Sprintf("%s GRPC call failed", methodName))
 		} else {
-			log.Info("GRPC call successful", "response", resp)
+			log.Info(fmt.Sprintf("%s GRPC call successful", methodName))
```
Comment (chrismuellner) [inline controllers/csi/driver/server.go:332]: is this log useful?
would removing the info log before the grpc call and just log error/success reduce log spam?

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: controllers/csi/driver/server.go (hunk 3 above)
Comment (0sewa0) [reply controllers/csi/driver/server.go:322]: I like `GRPC call`, it shows that the kubernetes made the request
it separates it from other logs

File: controllers/csi/driver/server.go (hunk 4 above)
Comment (0sewa0) [reply controllers/csi/driver/server.go:332]: there shouldn't be much spam anymore, but yeah I can't come up with a valid use for this :D

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**